docs/arcadia-users-group/20231104-testing-concepts/lesson.md

```diff
@@ -0,0 +1,573 @@
+# Testing concepts and terminology
+
+> Note that this lesson has been modified from [The Carpentries Incubator](https://github.com/carpentries-incubator) lesson on [Python Testing](https://carpentries-incubator.github.io/python-testing/).
+> Parts are reproduced in full, but the major changes were included to shorten the lesson to 60 minutes.
+
+Everyone tests their software to some extent, if only by running it and trying it out.
+Most programmers do a certain amount of exploratory testing while they write new code or change old code.
+This may involve printing or plotting outputs of your code as you program.
+
+Systematic testing codifies these behaviors, allowing them to be automatically applied quickly and repeatedly over entire code bases.
```

Comment:
I think in the introduction we should make it clearer why to test (making sure your function or program does what it is meant to do and handles different cases, making sure a change you introduce doesn't break things etc)

Reply:
oh good call. i modified the intro to add this, thank you!!